README.rst

```diff
@@ -829,6 +829,322 @@ Checking:
+Configure NGINX to serve encrypted traffic without server restarts when certificate or key changes occur by using `js_shared_dict_zone <https://nginx.org/en/docs/http/ngx_http_js_module.html#js_shared_dict_zone>`_ as a cache.
+
+Note: this example below work with njs >= `0.8.0 <http://nginx.org/en/docs/njs/changes.html#njs0.8.0>`_.
+
+This example demonstrtaes:
+
+ - Use of `js_set <https://nginx.org/en/docs/http/ngx_http_js_module.html#js_set>`_ in combination with ``ssl_certificate data:$var;`` to use NJS to resolve value of cert/key during handshake.
+ - Use of `js_shared_dict_zone <https://nginx.org/en/docs/http/ngx_http_js_module.html#js_shared_dict_zone>`_ to store cert/key in memory.
+ - Implementation a simple RESTful API to manage ``shared_dict`` to get/set certificate/key files.
+ - How to deal with ``Content-Disposition`` while handling file uploads in NJS.
+
+nginx.conf:
+
+.. code-block:: nginx
+
+  ...
+
+  load_module modules/ngx_http_js_module.so;
+  error_log /dev/stdout debug;
+  events {  }
+
+  http {
+    js_path "/etc/nginx/njs/";
+    js_import main from http/certs/js/dynamic.js;
+    js_shared_dict_zone zone=kv:1m;
+
+    server {
+      listen 80;
+      listen 443 ssl;
+      server_name www.example.com;
+
+      js_var $shared_dict_zone_name kv;
+      js_var $cert_folder '/tmp/';
+
+      js_set $dynamic_ssl_cert main.js_cert;
+      js_set $dynamic_ssl_key main.js_key;
+
+      ssl_password_file /etc/nginx/njs/http/certs/ca/password;
+      ssl_certificate data:$dynamic_ssl_cert;
+      ssl_certificate_key data:$dynamic_ssl_key;
+
+      location = / {
+        js_content main.info;
+      }
+
+      location /kv {
+        js_content main.kv;
+      }
+
+      location = /clear {
+        js_content main.clear_cache;
+      }
+    }
+
+  }
+
+
+example.js:
+
+.. code-block:: js
+
+  /**
+   * NGINX Secure Traffic Serving - JavaScript Module
+   *
+   * This JavaScript module enables NGINX to securely serve encrypted traffic without server restarts when certificate or key changes occur.
+   *
+   * Usage:
+   * 1. Install and configure NGINX with the NJS module.
+   * 2. Include the provided JavaScript module (dynamic.js) in your NGINX configuration.
+   * 3. Set up an HTTP endpoint to handle file uploads (e.g., /upload).
+   * 4. Clients can use the endpoint to upload certificate and key files using the 'curl' command, like so:
+   *    curl http://localhost:8000/upload -F cert=@/path/www.example.com.crt -F key=@/path/www.example.com.key
+   *
+   * Benefits:
+   * - Dynamic SSL certificate and key management without server restarts.
+   * - Handle certs/keys file uploads.
+   * - Efficient and uninterrupted serving of encrypted traffic using shared_dict to minimize disk IO and cache certs/keys.
+   *
+   * Note:
+   * - Ensure appropriate file permissions for the NGINX server to write uploaded files.
+   * - Validate and sanitize uploaded file content to prevent security risks.
+   */
+
+  import fs from 'fs'
```

Comment:
Because the is an example, I suggest to remove non-essential code from the .rst page, leave only crucial parts.

Reply:
please take a look again.